src/lib/ggpo/pevents.cpp

```diff
@@ -0,0 +1,603 @@
+                    result = pthread_mutex_unlock(&info->Waiter->Mutex);
+                    assert(result == 0);
+                    if (destroy) {
+                        info->Waiter->Destroy();
+                        delete info->Waiter;
+                    }
+                    continue;
+                }
+
+                if (info->Waiter->WaitAll) {
+                    --info->Waiter->Status.EventsLeft;
+                    assert(info->Waiter->Status.EventsLeft >= 0);
+                    // We technically should do i->Waiter->StillWaiting = Waiter->Status.EventsLeft
+                    // != 0 but the only time it'll be equal to zero is if we're the last event, so
+                    // no one else will be checking the StillWaiting flag. We're good to go without
+                    // it.
+                } else {
+                    info->Waiter->Status.FiredEvent = info->WaitIndex;
+                    info->Waiter->StillWaiting = false;
+                }
+
+                result = pthread_mutex_unlock(&info->Waiter->Mutex);
+                assert(result == 0);
+
+                result = pthread_cond_signal(&info->Waiter->CVariable);
+                assert(result == 0);
+            }
+            event->RegisteredWaits.clear();
+#endif // WFMO
+            result = pthread_mutex_unlock(&event->Mutex);
+            assert(result == 0);
+
+            result = pthread_cond_broadcast(&event->CVariable);
+            assert(result == 0);
+        }
+
+        return 0;
+    }
+
+    int ResetEvent(neosmart_event_t event) {
+        int result = pthread_mutex_lock(&event->Mutex);
+        assert(result == 0);
+
+        event->State = false;
+
+        result = pthread_mutex_unlock(&event->Mutex);
+        assert(result == 0);
+
+        return 0;
+    }
+
+#ifdef PULSE
+    int PulseEvent(neosmart_event_t event) {
+        // This may look like it's a horribly inefficient kludge with the sole intention of reducing
+        // code duplication, but in reality this is what any PulseEvent() implementation must look
+        // like. The only overhead (function calls aside, which your compiler will likely optimize
+        // away, anyway), is if only WFMO auto-reset waits are active there will be overhead to
+        // unnecessarily obtain the event mutex for ResetEvent() after. In all other cases (being no
+        // pending waits, WFMO manual-reset waits, or any WFSO waits), the event mutex must first be
+        // released for the waiting thread to resume action prior to locking the mutex again in
+        // order to set the event state to unsignaled, or else the waiting threads will loop back
+        // into a wait (due to checks for spurious CVariable wakeups).
+
+        int result = SetEvent(event);
+        assert(result == 0);
+        result = ResetEvent(event);
+        assert(result == 0);
+
+        return 0;
+    }
+#endif
+} // namespace neosmart
+
+#else //_WIN32
```

Comment:
We only include `pevents.h` for Unix-like platforms so we could just delete everything in this else clause.